certification/internal/shell/has_minimal_vulns.go

```diff
@@ -1,16 +1,133 @@
 package shell
 
 import (
+	"bytes"
+	"compress/bzip2"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+
 	"github.com/komish/preflight/certification"
-	"github.com/komish/preflight/certification/errors"
 	"github.com/sirupsen/logrus"
 )
 
+const (
+	ovalFilename = "rhel-8.oval.xml.bz2"
+	ovalUrl      = "https://www.redhat.com/security/data/oval/v2/RHEL8/"
+	reportFile   = "vuln.html"
+)
+
 type HasMinimalVulnerabilitiesCheck struct{}
 
 func (p *HasMinimalVulnerabilitiesCheck) Validate(image string, logger *logrus.Logger) (bool, error) {
-	return false, errors.ErrFeatureNotImplemented
+
+	ovalFileUrl := fmt.Sprintf("%s%s", ovalUrl, ovalFilename)
+
+	tempdir, err := os.MkdirTemp(os.TempDir(), "oval-*")
+	if err != nil {
+		logger.Error("Unable to create temp dir", err)
+		return false, err
+	}
+	logger.Debugf("Oval file dir: %s", tempdir)
+	defer os.RemoveAll(filepath.Dir(tempdir))
+
+	// TODO claen up the names
+	ovalFilePath := filepath.Join(tempdir, ovalFilename)
+	logger.Debugf("Oval file path: %s", ovalFilePath)
+
+	err = DownloadFile(ovalFilePath, ovalFileUrl)
+	if err != nil {
+		logger.Error("Unable to download Oval file", err)
+		return false, err
+	}
+	// get the file name
+	r := regexp.MustCompile(`(?P<filename>.*).bz2`)
+	ovalFilePathDecompressed := filepath.Join(tempdir, r.FindStringSubmatch(ovalFilename)[1])
+
+	err = Unzip(ovalFilePath, ovalFilePathDecompressed)
+	if err != nil {
+		logger.Error("Unable to unzip Oval file: ", err)
+		return false, err
+	}
+
+	numOfVulns, err := getNumberOfVulnerabilities(image, ovalFilePathDecompressed, logger)
+	if err != nil {
+		return false, err
+	}
+
+	logger.Debugf("The number of found vulnerabilities: %d", numOfVulns)
+	if numOfVulns > 0 {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func DownloadFile(filename string, url string) error {
```

Comment:
could be factored into a "utils"

Reply:
sounds like a great idea if I can ever make packages to work :/

Reply:
Done! Please let me know if it looks good :)

Reply:
nice looks good!